### DISCUSSION TITLE
`Uncaught Error: missing serializer: oZTRi-w3_` when trying to join a Room after upgrading

### DISCUSSION THREAD

**Discussion by luisoncpp** [colyseus/colyseus#583] [Q&A] [answered]
Hi everyone, I'm updating colyseus in a project and I'm getting an error by calling the following line:

```ts
    const room = await client.joinOrCreate<RoomStateType<TRoom>>(GetRoomId<TRoom>(room_spec), options);
```

The error is this one:

```
Protocol.ts:11 Uncaught Error: missing serializer: oZTRi-w3_
    at getSerializer (Protocol.ts:11:5)
    at Room.onMessageCallback (Room.ts:196:42)
```

I can see it in the Javascript console in the browser.

In server side, the Room is successfully calling `onJoin` with the corresponding object sent from the client in `options`.

The id of the serializer(`oZTRi-w3_` in this case) is always different in each run.

I'm upgrading from this version of colyseus:

```ts
		"@colyseus/command": "^0.1.6",
		"@colyseus/core": "^0.14.28",
		"@colyseus/mongoose-driver": "^0.14.22",
		"@colyseus/monitor": "^0.14.22",
		"@colyseus/ws-transport": "^0.14.21",
		"colyseus": "^0.14.24",
		"colyseus.js": "^0.14.1",
```

to this one:

```ts
		"@colyseus/command": "^0.2.1",
		"@colyseus/core": "^0.15.4",
		"@colyseus/mongoose-driver": "^0.15.0",
		"@colyseus/monitor": "^0.15.1",
		"@colyseus/ws-transport": "^0.15.0",
		"colyseus": "^0.15.0",
		"colyseus.js": "^0.15.0",
```

Any ideas what could be the issue?

Should I do anything extra before joining in colyseus 0.15 that wasn't required in colyseus 0.14?

**Comment by endel** [comment]
Hi @luisoncpp, this issue sounds like incompatible versions between client and server. Please check in your package-lock.json file if there are still some references to 0.14 - if there is try removing/updating that dependency.

Also If the problem persists try removing package-lock.lock and the node_modules folder, and running npm install again.

**Comment by luisoncpp** [comment, marked as answer]
Hi, I think I just solved the issue (unless it's one of those problems that randomly happen or stop happening), I guess the problem was that I had both `redis` and `ioredis` installed (I installed both because I had to update redis as well, I tried both APIs and forgot to delete the previous one).

**Reply by eugene-egorov** [reply]
Also faced this problem and yes this is because of outdated client version. Namely this line https://github.com/colyseus/colyseus.js/blob/5bb33ff85cb0943a4806a4cbb2d527e1d270e3ad/src/Room.ts#L193

**Reply by vinkent420** [reply]
Hi @luisoncpp , thank you for raising your issue, so am I. But I think root cause of "missing serializer" after upgrading is mismatch version of colyseus/schema between client and server, it requires same version (at least minor version), so to solve we need to upgrade version both client and server.

And for your reply above, it totally right, I also facing as same as you, haha. That is at new version of RedisDriver or RedisPrecense, they are using ioredis, and in own application I use also redis package for some specific logic, ioredis config option is different than redis (ex: host, port are separate - mean don't have uri property, db property at ioredis and database at redis..)

I really thank your reply as a hint myself @luisoncpp , I doubled check ioredis and updated in code a bit and it run!
